### PR TITLE
Fix the deprecation warning while connecting to Hadoop File System

### LINKED ISSUE #4905: hdfs: pyarrow.hdfs.connect is deprecated
## Bug Report
Since 2.0.0, `pyarrow.hdfs.connect` is deprecated, as seen on [this GHA log](https://github.com/iterative/dvc/runs/1415667578?check_suite_focus=true#step:7:3952), and recommends to use [`pyarrow.fs.HadoopFileSystem`](http://arrow.apache.org/docs/python/generated/pyarrow.fs.HadoopFileSystem.html).

```
home/runner/work/dvc/dvc/dvc/tree/pool.py:53: DeprecationWarning: pyarrow.hdfs.connect is deprecated as of 2.0.0, please use pyarrow.fs.HadoopFileSystem instead.
    return self._conn_func(*self._conn_args, **self._conn_kwargs)
```

Here are the usages:
https://github.com/iterative/dvc/blob/e1b82c5222930c55886ca16a48c3a223d05b4af0/dvc/tree/hdfs.py#L51-L56
https://github.com/iterative/dvc/blob/ca2e6a004cb6268c7b62f79ba22b55c5b81d8984/tests/remotes/hdfs.py#L20
https://github.com/iterative/dvc/blob/ca2e6a004cb6268c7b62f79ba22b55c5b81d8984/tests/remotes/hdfs.py#L128

### PR DESCRIPTION
**Pull request by kgritesh**
Fixes #4905

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/tree/hdfs.py [hunk 1]
```diff
@@ -46,10 +46,11 @@ def __init__(self, repo, config):
 
     @staticmethod
     def hdfs(path_info):
+
```
Comment (efiop) [inline dvc/tree/hdfs.py:49]:
```suggestion
```

**Review by efiop — APPROVED** [review]
Thank you so much @kgritesh! :pray:

**Comment by efiop** [comment]
@kgritesh Btw, please go to https://github.com/settings/emails and add the email that you've used to create your commit. Right now github is not able to link your commits to your github profile.

**Review by skshetry — CHANGES_REQUESTED** [review]
This still does not fix the issue. Please see this [log on the Actions](https://github.com/iterative/dvc/pull/4936/checks?check_run_id=1439560851#step:7:4036).

**Comment by efiop** [comment]
Good catch @skshetry ! :pray:

Closing this, since there is clearly a lot more work to be done and @jwokaty has already been working on a correct fix.

**Pull request closed without merging**